### PR TITLE
Let event loop threads implement reactor NonBlocking

### PR DESCRIPTION
**Pull request by yawkat**
This patch updates the default netty ThreadFactory to have the threads implement reactor NonBlocking. Any reactor blocking operations will then throw when they're done on an event loop thread.

### CODE REVIEW

File: http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactoryConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @param reactorNonBlocking If {@code true} (the default), netty event loop threads will implement
+ *                           project reactor {@link reactor.core.scheduler.NonBlocking} by default.
+ *                           This will make reactor blocking operations throw an exception.
+ * @since 4.4.0
+ * @author Jonas Konrad
+ */
+@ConfigurationProperties("netty.default.thread-factory")
+@BootstrapContextCompatible
+@Internal
+public record NettyThreadFactoryConfiguration(
```
Comment (graemerocher) [inline http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactoryConfiguration.java]: do we need a whole new configuration properties class for this? Can it not go in `NettyGlobalConfiguration`?

Comment (yawkat) [reply http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactoryConfiguration.java]: sure

**Comment by yawkat** [comment]
this will likely break security. in the meeting i suggested we move the security filter to blocking for now.

**Comment by graemerocher** [comment]
security is not implemented correctly if it is blocking and if it is blocking needs to be declared as such.

**Review by graemerocher — APPROVED** [review]

File: http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactory.java [hunk 2]
```diff
@@ -60,6 +63,28 @@ public class NettyThreadFactory {
...
 
+    private final NettyGlobalConfiguration configuration;
+
+    /**
+     * Create a new netty ThreadFactory factory.
+     *
+     * @param configuration The configuration
+     * @since 4.4.0
+     */
+    public NettyThreadFactory(NettyGlobalConfiguration configuration) {
```
Comment (sdelamo) [inline http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactory.java:76]: should not we annotate the new constructor with `@Inject`?

**Review by sdelamo — CHANGES_REQUESTED** [review]
I created several suggestions: 

- https://github.com/micronaut-projects/micronaut-core/pull/10622
- https://github.com/micronaut-projects/micronaut-core/pull/10621
- https://github.com/micronaut-projects/micronaut-core/pull/10620

File: http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactory.java (hunk 2 above)
Comment (sdelamo) [reply http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactory.java:76]: see: https://github.com/micronaut-projects/micronaut-core/pull/10621

**Review by sdelamo — APPROVED** [review]

**Pull request merged**